src/core/Platform.cpp

```diff
@@ -0,0 +1,116 @@
+///////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2008-2017, Sony Pictures Imageworks Inc
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimer.
+// Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+// Neither the name of the organization Sony Pictures Imageworks nor the
+// names of its contributors
+// may be used to endorse or promote products derived from this software
+// without specific prior written permission.
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS
+// IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+// TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+// PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER
+// OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+///////////////////////////////////////////////////////////////////////////////
+
+#include <OpenColorIO/OpenColorIO.h>
+
+#include "Platform.h"
+
+OCIO_NAMESPACE_ENTER
+{
+
+    namespace Platform
+    {
+        char* getenv (const char* name)
+        {
+#ifdef WINDOWS
+            static std::string str;
+
+            char * value;
+            size_t len = 0;         
+            const errno_t err = ::_dupenv_s(&value, &len, name);
+            if(err!=0 || len==0 || !value || !*value)
+            {
+                if(value) free(value);
+                return 0x0;
+            }
+            else
+            {
+                str.resize(len+1);
+                ::snprintf(&str[0], len, "%s", value);
+                if(value) free(value);
+            }
+
+            return const_cast<char*>(str.c_str());

```

Comment:
The use of this static is not thread safe, it will be a problem if there's any chance that Platform::getenv() is called simultaneously by two threads.

I'm a little confused about why this function is needed. I just use getenv even on Windows, for OIIO/OSL. Am I doing something dangerous?

This function should contain a comment with some discussion about exactly why _dupenv_s should be preferred over getenv (it wasn't immediately apparent to me even looking at the MSDN pages for both functions).